File: flutter_frontend/lib/models/attendance.dart
class Attendance {
  final int id;
  final DateTime checkInTime;
  final DateTime? checkOutTime;
  final String status;
  final double confidence;
  final String? imagePath;

  Attendance({
    required this.id,
    required this.checkInTime,
    this.checkOutTime,
    required this.status,
    required this.confidence,
    this.imagePath,
  });

  factory Attendance.fromJson(Map<String, dynamic> json) {
    return Attendance(
      id: json['id'],
      checkInTime: DateTime.parse(json['check_in_time']),
      checkOutTime: json['check_out_time'] != null
          ? DateTime.parse(json['check_out_time'])
          : null,
      status: json['status'],
      confidence: json['confidence'].toDouble(),
      imagePath: json['image_path'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'check_in_time': checkInTime.toIso8601String(),
      'check_out_time': checkOutTime?.toIso8601String(),
      'status': status,
      'confidence': confidence,
      'image_path': imagePath,
    };
  }
} 

File: backend/models/attendance.py
from datetime import datetime
from .user import db

class Attendance(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
    check_in_time = db.Column(db.DateTime, default=datetime.utcnow)
    check_out_time = db.Column(db.DateTime, nullable=True)
    status = db.Column(db.String(20), default='present')  # present, late, absent
    location = db.Column(db.String(100), nullable=True)
    image_path = db.Column(db.String(255), nullable=True)  # Path to stored face image
    confidence = db.Column(db.Float, nullable=True)  # Face recognition confidence score